pkg/controller/oam/applicationconfiguration/apply.go

```diff
@@ -19,61 +19,210 @@ package applicationconfiguration
 import (
 	"context"
 
-	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
 	"github.com/crossplane/crossplane-runtime/pkg/fieldpath"
 	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane/oam-kubernetes-runtime/apis/core/v1alpha2"
+
+	"github.com/pkg/errors"
 )
 
 // Reconcile error strings.
 const (
 	errFmtApplyWorkload  = "cannot apply workload %q"
 	errFmtSetWorkloadRef = "cannot set trait %q reference to %q"
 	errFmtApplyTrait     = "cannot apply trait %q"
+	errFmtApplyScope     = "cannot apply scope %q"
 )
 
 // A WorkloadApplicator creates or updates workloads and their traits.
 type WorkloadApplicator interface {
 	// Apply a workload and its traits.
-	Apply(ctx context.Context, w []Workload, ao ...resource.ApplyOption) error
+	Apply(ctx context.Context, namespace string, status []v1alpha2.WorkloadStatus, w []Workload, ao ...resource.ApplyOption) error
```

Comment:
Why change the stub?

Reply:
I need the status of the workloads to calculate a diff. This allows me to remove workloadRefs from health scope.